### ISSUE TITLE
"HTTPS endpoint unresponsive and insecure mode isn't enabled" with boot2docker

### ISSUE THREAD

**Issue by ggtools** [docker/compose#1673]
**Foreword**

Since issue #563 is closed I'm opening a new one, @GordonTheTurtle feel free to close it and re-open the previous one.

**Environment**

Docker 1.7.0, compose 1.3.1 on MacOSX with Boot2docker from Homebrew

Does not happen on Ubuntu 14_04.

**How to reproduce**

From a minimal `docker-compose.yml`:

```yaml
server:
    image: jfrog-docker-registry.bintray.io/artifactory/artifactory-oss
```

`docker-compose up` does not work:

```
/usr/local/Cellar/fig/1.3.1/libexec/vendor/lib/python2.7/site-packages/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Pulling server (jfrog-docker-registry.bintray.io/artifactory/artifactory-oss:latest)...
Traceback (most recent call last):
  File "/usr/local/Cellar/fig/1.3.1/libexec/bin/docker-compose", line 9, in <module>
    load_entry_point('docker-compose==1.3.1', 'console_scripts', 'docker-compose')()
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/cli/main.py", line 31, in main
    command.sys_dispatch()
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/cli/docopt_command.py", line 21, in sys_dispatch
    self.dispatch(sys.argv[1:], None)
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/cli/command.py", line 34, in dispatch
    super(Command, self).dispatch(*args, **kwargs)
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/cli/docopt_command.py", line 24, in dispatch
    self.perform_command(*self.parse(argv, global_options))
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/cli/command.py", line 66, in perform_command
    handler(project, command_options)
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/cli/main.py", line 460, in up
    do_build=not options['--no-build'],
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/project.py", line 230, in up
    do_build=do_build,
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/service.py", line 320, in execute_convergence_plan
    do_build=do_build,
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/service.py", line 208, in create_container
    insecure_registry=insecure_registry,
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/service.py", line 236, in ensure_image_exists
    self.pull(insecure_registry=insecure_registry)
  File "/usr/local/Cellar/fig/1.3.1/libexec/lib/python2.7/site-packages/compose/service.py", line 693, in pull
    insecure_registry=insecure_registry)
  File "/usr/local/Cellar/fig/1.3.1/libexec/vendor/lib/python2.7/site-packages/docker/client.py", line 883, in pull
    repository, insecure=insecure_registry
  File "/usr/local/Cellar/fig/1.3.1/libexec/vendor/lib/python2.7/site-packages/docker/auth/auth.py", line 59, in resolve_repository_name
    return expand_registry_url(parts[0], insecure), parts[1]
  File "/usr/local/Cellar/fig/1.3.1/libexec/vendor/lib/python2.7/site-packages/docker/auth/auth.py", line 38, in expand_registry_url
    "HTTPS endpoint unresponsive and insecure mode isn't enabled."
docker.errors.DockerException: HTTPS endpoint unresponsive and insecure mode isn't enabled.
```

The problem does not happen if the image is pulled before using `docker pull jfrog-docker-registry.bintray.io/artifactory/artifactory-oss``

**Comment by aanand** [comment]
ping @shin- - I've seen people report this a few times on docker-py before. What's the source of the problem?

**Comment by jwhitcraft** [comment]
I'm seeing this too. I have a 2.0.1 registry running behind our firewall and it has SSL enabled, but when I try and run the docker-compose up, it just spits out the same error as the OP.  docker pull works fine.

**Comment by streamnsight** [comment]
Seen the same problem, even though my registry is using a signed certificate and the CA cert is on the client side too, which had been a problem when pushing / pulling images with docker. Having the CA cert on the client solved those problems, but even though docker has no problem pulling the images, compose still complains.

**Comment by dnephin** [comment]
This should be fixed by #1736

**Comment by aanand** [comment]
Fixed by #1736 and released in [Compose 1.4.0 RC2](https://github.com/docker/compose/releases/tag/1.4.0rc2).
